lib/src/map/flutter_map_state.dart

```diff
@@ -756,15 +756,16 @@ class FlutterMapState extends MapGestureMixin
   double _calculateScreenHeightInDegrees() =>
       options.screenSize!.height * 170.102258 / math.pow(2, zoom + 8);
 
-  static FlutterMapState? maybeOf(BuildContext context, {bool nullOk = false}) {
-    final widget =
-        context.dependOnInheritedWidgetOfExactType<MapStateInheritedWidget>();
-    if (nullOk || widget != null) {
-      return widget?.mapState;
-    }
-    throw FlutterError(
-        'MapState.of() called with a context that does not contain a FlutterMap.');
-  }
+  static FlutterMapState? maybeOf(BuildContext context) => context
+      .dependOnInheritedWidgetOfExactType<MapStateInheritedWidget>()
+      ?.mapState;
+
+  static FlutterMapState of(BuildContext context) =>
+      context
+          .dependOnInheritedWidgetOfExactType<MapStateInheritedWidget>()
+          ?.mapState ??
+      (throw StateError(
+          '`FlutterMapState.of()` should not be called outside a `FlutterMap` and its children'));
 }
```

Comment:
You can directly call `maybeOf(context)` to avoid code duplication (I also find it clearer to use a `if` to throw an error and beneficiate of type shadowing but it's a personal preference):

```dart
static FlutterMapState of(BuildContext context) {
  final state = maybeOf(context);
  
  if (state == null) {
    throw StateError('...');
  }

  return state;
}
```

Otherwise the rest looks good to me 👍

Reply:
Good catch, I was intending to use `maybeOf` in `of`, but I got sidetracked.
I'll keep using `??`, as it's just shorter and similar in clarity (if you're familiar with Dart).

Fixed in fc56357.